stdlib/public/Concurrency/Series.swift

```diff
@@ -0,0 +1,638 @@
+///       print(digit)
+///     }
+///
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+public struct Series<Element> {
+  public struct Continuation: Sendable {
+    fileprivate let storage: _SeriesBufferedStorage<Element, Never>
+
+    /// Resume the task awaiting the next iteration point by having it return
+    /// nomally from its suspension point or buffer the value if no awaiting
+    /// next iteration is active.
+    ///
+    /// - Parameter value: The value to yield from the continuation.
+    ///
+    /// This can be called more than once and returns to the caller immediately
+    /// without blocking for any awaiting consuption from the iteration.
+    public func resume(yielding value: __owned Element) {
+      storage.yield(value)
+    }
+
+    /// Resume the task awaiting the next iteration point by having it return
+    /// nil which signifies the end of the iteration.
+    ///
+    /// Calling this function more than once is idempotent; i.e. finishing more
+    /// than once does not alter the state beyond the requirements of
+    /// AsyncSequence; which claims that all values past a terminal state are
+    /// nil.
+    public func finish() {
+      storage.yield(nil)
+    }
+
+    /// A callback to invoke when iteration of a Series is cancelled.
+    ///
+    /// If an `onCancel` callback is set, when iteration of a Series is
+    /// cancelled via task cancellation that callback is invoked. The callback
+    /// is disposed of after any terminal state is reached.
+    ///
+    /// Cancelling an active iteration will first invoke the onCancel callback
+    /// and then resume yeilding nil. This means that any cleanup state can be
+    /// emitted accordingly in the cancellation handler
+    public var onCancel: (@Sendable () -> Void)? {
+      get {
+        return storage.onCancel
+      }
+      nonmutating set {
+        storage.onCancel = newValue
+      }
+    }
+  }
+
+  let produce: (UnsafeContinuation<Element?, Never>) -> Void
+  let cancel: @Sendable () -> Void
+
+  /// Construct a Series buffering given an Element type.
+  ///
+  /// - Parameter buffering: The type the Series will produce.
+  /// - Parameter maxBufferedPendingElements: The maximum number of elements to
+  ///   hold in the buffer past any checks for continuations being resumed.
+  /// - Parameter build: The work associated with yielding values to the Series.
+  ///
+  /// The maximum number of pending elements limited by dropping the oldest
+  /// value when a new value comes in if the buffer would excede the limit
+  /// placed upon it. By default this limit is unlimited.
+  ///
+  /// The build closure passes in a Continuation which can be used in
+  /// concurrent contexts. It is thread safe to send and finish; all calls are
+  /// to the continuation are serialized, however calling this from multiple
+  /// concurrent contexts could result in out of order delivery.
+  public init(
+    buffering: Element.Type = Element.self,
```

Comment:
I think we were going to remove the default value here — looks like you took it out of `ThrowingSeries` already.

Reply:
I am really torn on that; it really ends up being quite nice to write the return values... e.g.

```swift
var someSeriesOfThings: Series<Thing> {
   Series { continuation in
      ...
   }
}
```